### PR TITLE
CUMULUS-2986: Lambda Memory Sizes in Terraform

### PR DESCRIPTION
**Pull request by Nnaga1**
**Summary:** Summary of changes

-Added the option to configure lambda memory sizes with lambda functions that have customizable timeouts enabled.
Addresses [CUMULUS-2986: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2986)

## Changes

* Detailed list or prose of changes
Added lambda_memory_sizes to variables.tf of terraform modules, and also changed the initialization of memory sizes to look in the lambda_memory_sizes array, which can be configured in terraform.tfvars of cumulus-tf. 

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

**Comment by chuckwondo** [comment]
@Nnaga1, I suspect you didn't intent to commit the 2 large zip files.

**Comment by chuckwondo** [comment]
> yes I'm not sure why those were added, looking into getting rid of them now. Essentially all I'm trying to add is the memory_sizes fields into the terraform files (variables.tf and the files that use them).

Perhaps you used `git add .` (with a dot [`.`], rather than specific files) to add files?

File: docs/configuration/task-configuration.md [hunk 1]
```diff
@@ -88,3 +88,36 @@ lambda_timeouts = {
...
+## lambda_memory_sizes
+
+A configurable map of memory sizes (in MBs) for cumulus ingest module task lambdas in the form:
+
+```hcl
+<lambda_identifier>_memory_size: <memory_size>
+  type = map(string)
+```
+
+Currently the following values are supported:
```
Comment (Jkovarik) [inline docs/configuration/task-configuration.md:101]: ~Was files_to_granules left out?   It's not *really* a production lambda in that I wouldn't expect most users are utilizing it as it's really used to make test/example workflows work, but it *is* a deployed resource that comes with Core. 

There's a sensible argument for us moving it to the example project and template repo, but that's probably a separate convo. 

Same thing for fake-processing-task~


*edit nevermind, I see we also didn't add those to the timeout block, and probably had that same discussion*      I'm going to leave this alone in favor of an arch discussion.

Comment (Jkovarik) [reply docs/configuration/task-configuration.md:101]: What is missing, however is `add_misssing_checksums` and `update-cmr-access-constraints`.   That's not an issue directly with this PR or this work, but we should probably add them to this and the timeout list.

**Review by Jkovarik — CHANGES_REQUESTED** [review]
Great work! 

 I validated the changes work for the listed lambdas and/or looked for anything missing.  There's a couple of comments/minor issues for your review/commentary, let me know what you think.

File: tf-modules/ingest/post-to-cmr-task.tf [hunk 2]
```diff
@@ -6,7 +6,7 @@ resource "aws_lambda_function" "post_to_cmr_task" {
   role             = var.lambda_processing_role_arn
   runtime          = "nodejs14.x"
   timeout          = lookup(var.lambda_timeouts, "post_to_cmr_task_timeout", 300)
-  memory_size      = 256
+  memory_size      = lookup(var.lambda_memory_sizes, "post_to_cmr_task_memory_size", 256)
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply tf-modules/ingest/post-to-cmr-task.tf:9]: I think 512 is probably a good improvement for this lambda, and defensible generally.       If we do that we need to note it in the changelog.

File: CHANGELOG.md [hunk 3]
```diff
@@ -21,8 +23,32 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
       - queue_pdrs_task_memory_size
       - queue_workflow_task_memory_size
       - sync_granule_task_memory_size
+      - update_cmr_access_constraints_task_memory_size
       - update_granules_cmr_task_memory_size
   - Initializes the lambda_memory_size(s) variable in the Terraform variable list
+  
+=======
```
Comment (Jkovarik) [inline CHANGELOG.md:30]: This changeset doesn't look correct.

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,28 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 
+<<<<<<< HEAD
```
Comment (Jkovarik) [inline CHANGELOG.md:9]: This looks like a bad merge, please fix!

**Review by Jkovarik — DISMISSED** [review]
Thanks for addressing my feedback -- there's still a merge issue that needs sorted, then this looks good to go.

File: CHANGELOG.md [hunk 5]
```diff
@@ -45,6 +65,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ## [v13.3.0] 2022-8-19
 
+>>>>>>> master
```
Comment (kkelly51) [inline CHANGELOG.md:68]: oop delete this line

**Review by kkelly51 — APPROVED** [review]
Looks good!

**Pull request merged**